.github/actions/tag-release/action.yml

```diff
@@ -0,0 +1,86 @@
+# Workflow to create a new tag release when a release branch is merged
+#
+# Recommended for caller to support triggering on both `pull_request` merge and `workflow_dispatch` for flexibility
+#
+# TODO: Add example workflows
+name: Tag release
+
+description: Bump crate versions and open release PR
+
+inputs:
+  release-branch:
+    description: 'Branch to release'
+    required: true
+  version:
+    description: 'SemVer release version'
+    required: true
+  tag-prefix:
+    description: 'Tag prefix used to get most recent release'
+    required: false
+    default: ""
+  changelog-path:
+    description: 'Relative path to look for changelog'
+    required: false
+    default: "."
+  changelog-config-file:
+    description: 'Relative path to the config file for `mikepenz/release-changelog-builder-action`'
+    required: false
+    default: ""
+
+runs:
+  using: "composite"
+  steps:
+    - name: Get version
+      id: get-version
+      run: |
+        if [[ -n "${{ inputs.tag-prefix }}" ]]; then
+          TAG_PREFIX="${{ inputs.tag-prefix }}-"
+        else
+          TAG_PREFIX=""
+        fi
+
+        TAG_VERSION=${TAG_PREFIX}v${{ inputs.version }}
+
+        git tag -a $TAG_VERSION -m "$TAG_VERSION" origin/${{ inputs.release-branch }}
+        git push origin $TAG_VERSION --follow-tags
+        echo "tag-version=$TAG_VERSION" | tee -a "$GITHUB_OUTPUT"
+      shell: bash
+
```

Comment:
I would consider either testing the length of the `get-version` or making the subsequent steps conditional on `if: steps.get-version.outputs.tag-version != ''`